addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -2331,6 +2346,50 @@ export class OdooEditor extends EventTarget {
                 }
             }
         } else if (node.nodeType !== Node.TEXT_NODE) {
+            if (node.nodeName === 'TD' && node.hasAttribute('style')) {
+                // TD tags do not support style in v15, move style to children or new SPAN.
+                const spans = node.childNodes.length === 0 || [...node.childNodes].some(n => !['SPAN', 'FONT'].includes(n.nodeName))
+                    ? [this.document.createElement('SPAN')]
+                    : [...node.childNodes];
+                node.style.backgroundColor = null; // background-color: on TD (cell) != on span (text).
+                spans.forEach(span => {
+                    node.getAttribute('style').split(';').forEach(styleText => {
+                        // Give parent's style to span child, unless it already exists.
+                        const [styleName, styleValue] = styleText.split(':').map(x => x.trim());
+                        if (!span.style[styleName]) {
+                            span.style[styleName] = styleValue;
+                        }
+                    });
+                    if (span.parentNode !== node) {
+                        // A new span was needed to apply the styles. Use it to wrap the nodes in the cell.
+                        span.append(...node.childNodes);
+                        node.append(span);
+                    }
+                    this._cleanForPaste(span);
+                })
+            } else if (node.nodeName === 'FONT') {
+                // FONT tags have some style information in custom attributes,
+                // this maps them to the style attribute.
+                const attributesToStyle = { 'color': 'color', 'size': 'font-size'};
+                for (const [attributeName, styleName] of Object.entries(attributesToStyle)) {
+                    if (node.hasAttribute(attributeName) && !node.style[styleName]) {
+                        node.style[styleName] = node.getAttribute(attributeName);
+                        if (styleName === 'font-size' && !node.style[styleName]) {
+                            // FONT size uses non-standard numeric values
+                            node.style[styleName] = +node.getAttribute(attributeName) + 10 + 'pt';
+                        }
+                    }
+                }
+            } else if (['S', 'U'].includes(node.nodeName) && node.childNodes.length === 1 && node.firstChild.nodeName === 'FONT') {
+                // S and U tags sometimes contain FONT tags. We prefer the
+                // strike to adopt the style of the text, so we invert them.
+                const fontNode = node.firstChild;
+                node.parentNode.appendChild(fontNode);
```

Comment:
This is not correct because `node.parentNode` might have more children than just `node`. I think what you want is to insert `fontNode` right where `node` is, as you'll remove `node` later on by inserting it back into `fontNode`, so you can do `node.before(fontNode)` (or `after`).

Reply:
Excellent !